block/internal/submitting/da_submitter.go

```diff
@@ -557,6 +583,7 @@ func marshalItems[T any](
 	}
 
 	// Wait for all goroutines to complete and check for errors
+	defer wg.Wait()
```

Comment:
Why did you add the wait group? The code below was waiting for all Go routines to land on the happy path and exited early on failures.